app/routes/animal.py

```diff
@@ -56,6 +58,26 @@ def update_one_animal(animal_id):
     
     return animal_to_update.to_dict(), 200
 
+@animals_bp.route("/<animal_id>", methods=["PATCH"])
+def patch_one_animal(animal_id):
+    animal_to_update = get_valid_item_by_id(Animal, animal_id)
+    request_body = request.get_json()
+
+    # Get all the column keys from Animal, and see if they're in the request.
+    # If they are, update the values of animal_to_update using values from the request.
+    mapper = inspect(Animal)
+    for column in mapper.attrs:
+        key = column.key
+        if key in request_body:
+            setattr(animal_to_update, key, request_body[key])
```

Comment:
I think this pattern is a little different than what we've seen in class.

Is this the case:

- This pattern is just as valid as the methods we've seen before (in class)
- This pattern is just cool, regular Python code, and a nice way to approach the problem using `inspect` and reading the `request_body`, and isn't necessarily more Pythonic/Flask

?

Or is this a pattern that I should encourage students to emulate?

Either way, I like it and approve :D TY for this work

Reply:
I don't know if it's more pythonic or flasky! I think it's RESTful, but there are also recommendations to make routes for each key instead -- e.g. `PATCH /animals/<id>/species`. So that's an alternative option.

I saw some students use a pattern like this for PUT (e.g. [this PR](https://github.com/Ada-C19/task-list-api/pull/33/files#diff-f67826701212aab477be0634a23fdcd7ffdfe748b8ce35eb27b8f690d334c732R69-R76)), and riffed off of that.